src/treecontroller/viewconsumable.js

```diff
@@ -0,0 +1,486 @@
+		if ( this.test( element, consumables ) ) {
+			this._consumables.get( element ).consume( consumables );
+
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Reverts provided {@link engine.treeView.Element view Element} and its parts so they can be consumed once again.
+	 * Method does not revert items that were never previously added for consumption, even if they are included in
+	 * consumables object.
+	 *
+	 *		viewConsumable.revert( p, { name: true } ); // Reverts element's name.
+	 *		viewConsumable.revert( p, { attribute: 'name' } ); // Reverts element's attribute.
+	 *		viewConsumable.revert( p, { class: 'foobar' } ); // Reverts element's class.
+	 *		viewConsumable.revert( p, { style: 'color' } ); // Reverts element's style.
+	 *		viewConsumable.revert( p, { attribute: 'name', style: 'color' } ); // Reverts attribute and style.
+	 *		viewConsumable.revert( p, { class: [ 'baz', 'bar' ] } ); // Multiple names can be reverted.
+	 *
+	 * Throws {@link utils.CKEditorError CKEditorError} `viewconsumable-invalid-attribute` when `class` or `style`
+	 * attribute is provided - it should be handled separately by providing actual style/class.
+	 *
+	 *		viewConsumable.revert( p, { attribute: 'style' } ); // This call will throw an exception.
+	 *		viewConsumable.revert( p, { style: 'color' } ); // This is properly handled style.
+	 *
+	 * @param {engine.treeView.Element} element
+	 * @param {Object} consumables
+	 * @param {Boolean} consumables.name If set to true element's name will be included.
+	 * @param {String|Array} consumables.attribute Attribute name or array of attribute names.
+	 * @param {String|Array} consumables.class Class name or array of class names.
+	 * @param {String|Array} consumables.style Style name or array of style names.
+	 */
+	revert( element, consumables ) {
+		const elementConsumables = this._consumables.get( element );
+
+		if ( elementConsumables !== undefined ) {
+			elementConsumables.revert( consumables );
+		}
+	}
+
+	/**
+	 * Creates consumable object from {@link engine.treeView.Element view Element}. Consumable object will include
+	 * element's name and all its attributes, classes and styles.
+	 *
+	 * @static
+	 * @param {engine.treeView.Element} element
+	 * @returns {Object} consumables
+	 */
+	static consumablesFromElement( element ) {
+		const consumables = {
+			name: true,
+			attribute: [],
+			class: [],
+			style: []
+		};
+
+		const attributes = element.getAttributeKeys();
+
+		for ( let attribute of attributes ) {
+			// Skip classes and styles - will be added separately.
+			if ( attribute == 'style' || attribute == 'class' ) {
+				continue;
+			}
+
+			consumables.attribute.push( attribute );
+		}
+
+		const classes = element.getClassNames();
+
+		for ( let className of classes ) {
+			consumables.class.push( className );
+		}
+
+		const styles = element.getStyleNames();
+
+		for ( let style of styles ) {
+			consumables.style.push( style );
+		}
+
+		return consumables;
+	}
```

Comment:
Additional static method would be useful which takes an element or document fragment and build instance of a  `ViewConsumable` class with all elemenets in the tree.